### PR TITLE
docs: improve doc for server timeout

### PR DESCRIPTION
**Pull request by yichengq**
@xiang90

### CODE REVIEW

File: Documentation/2.0/admin_guide.md [hunk 1]
```diff
@@ -162,20 +162,21 @@ etcd sets different timeouts for various types of client requests. The timeout v
 
 #### Get requests
 
-No timeout is set for get requests, because etcd can always return the result in a short time.
+Timeout is not set for get requests, because etcd serves the result locally in a short time.
```
Comment (xiang90) [inline Documentation/2.0/admin_guide.md]: The main reason is that it is also non-blocking. For example, zk block read even though it serves locally to ensure local ordering.

File: Documentation/2.0/admin_guide.md [hunk 2]
```diff
@@ -162,20 +162,21 @@ etcd sets different timeouts for various types of client requests. The timeout v
...
+
+*Note*: QuorumGet request is a different type, which is mentioned in the following sections.
```
Comment (xiang90) [inline Documentation/2.0/admin_guide.md]: double stars

File: Documentation/2.0/admin_guide.md [hunk 3]
```diff
@@ -162,20 +162,21 @@ etcd sets different timeouts for various types of client requests. The timeout v
...
 
 #### Watch requests
 
-Unlimited timeout is set for watch requests. Clients can keep long-polling watch requests as long as they want.
+Unlimited timeout is set for watch requests. etcd will not stop a watch request until client cancels it.
```
Comment (xiang90) [inline Documentation/2.0/admin_guide.md]: Timeout is not set for ...

**Comment by yichengq** [comment]
updated

File: Documentation/admin_guide.md [hunk 4]
```diff
@@ -185,20 +185,21 @@ etcd sets different timeouts for various types of client requests. The timeout v
...
 
-No timeout is set for get requests, because etcd can always return the result in a short time.
+Timeout is not set for get requests, because etcd serves the result locally in a non-blocking way.
+
+**Note**: QuorumGet request is a different type, which is mentioned in the following sections.
 
 #### Watch requests
 
-Unlimited timeout is set for watch requests. Clients can keep long-polling watch requests as long as they want.
+Timeout is not set for watch requests. etcd will not stop a watch request until client cancels it.
```
Comment (xiang90) [inline Documentation/admin_guide.md]: or the connection is broken

File: Documentation/admin_guide.md [hunk 5]
```diff
@@ -185,20 +185,21 @@ etcd sets different timeouts for various types of client requests. The timeout v
...
 
 #### Delete, Put, Post, QuorumGet requests
 
-The default timeout is large enough to allow all key modifications in a healthy cluster, even when it is electing a new leader.
+The default timeout(around 5s) is large enough to allow all key modifications if the majority of cluster is functioning.
```
Comment (xiang90) [inline Documentation/admin_guide.md]: The default timeout is 5 seconds. It should be ...

**Comment by xiang90** [comment]
lgtm

**Pull request merged**